Assignment4/JulioMartinezB-e22a329/Task7.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+Automatically generated by Colaboratory.
+Original file is located at
+    https://colab.research.google.com/drive/1tV5j-DRcpPtoJGoMj8v2DSqR_9wyXeiE
+**Task 07: Querying RDF(s)**
+"""
+
+"!pip install rdflib "
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2020-2021/master/Assignment4"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/resources/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+#RDFLib
+print("")
+print("") 
+print("Task 7.1")
+print("")
+print("RDFLib:")
+ns=Namespace("http://somewhere#")
+for subj,pred,obj in g.triples((None, RDFS.subClassOf,ns.Person)):
+    print(subj)
+
+# SPARQL 
+print("")
+print("SPARQL:")
+from rdflib.plugins.sparql import prepareQuery
+
+q1 = prepareQuery('''
+  SELECT 
+    ?subj
+  WHERE { 
+    ?subj  rdfs:subClassOf/rdfs:subClassOf*  ns:Person. 
+  }
+  ''', initNs = { "ns": ns,"rdfs":RDFS}
+)
+for p in g.query(q1):
+    print(p)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**"""
+
+#RDFLib
+print("")
+print("")
+print("Task 7.2")
+print("")
+print("RDFLib:")
+for subj in g.subjects(RDF.type,ns.Person):
+    print(subj)
+  
+
+for subj in g.subjects(RDF.type/RDFS.subClassOf,ns.Person):
```

Comment:
I do not think this is a valid statement in rdflib. It will be interpreted as a division. Have you run this?

Reply:
I ran it and it worked. I have now checked with another solution based (commented code) on what a classmate told me and both give the same output. I have changed to that solution.